service/history/shardController.go

```diff
@@ -391,6 +391,10 @@ func (i *historyShardsItem) getOrCreateEngine(shardClosedCh chan<- int) (Engine,
 		if err != nil {
 			return nil, err
 		}
+		if context.ShardOwnershipChanged() {
+			i.GetMetricsClient().RecordTimer(metrics.ShardInfoScope, metrics.ShardItemAcquisitionLatency,
+				context.GetCurrentTime(i.GetClusterMetadata().GetCurrentClusterName()).Sub(context.GetLastUpdatedTime()))
+		}
```

Comment:
I am not sure if this can give you accurate value for shard_ownership_change time - UpdateShard is called only every minute unless we run out of rangeID space, in which case, it is immediately invoked.

Reply:
Yep this is an approximation, I plan to also add code to ensure that shards are updated during shutdown to ensure most up to date timestamps.